### PR TITLE
Feature/spear bits audit1

### PR DESCRIPTION
**Pull request by invocamanman**
This PR depends on https://github.com/0xPolygonHermez/zkevm-contracts/pull/81
This PR solves the issues: 

https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/72

https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/69

https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/58

https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/59

https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/55

### CODE REVIEW

**Review by laisolizq — APPROVED** [review]

**Review by ignasirv — APPROVED** [review]
💯

File: contracts/PolygonZkEVMBridge.sol [hunk 1]
```diff
@@ -587,35 +578,29 @@ contract PolygonZkEVMBridge is
      * @param index Index
      */
     function isClaimed(uint256 index) public view returns (bool) {
-        uint256 claimedWordIndex = index / 256;
-        uint256 claimedBitIndex = index % 256;
-        uint256 claimedWord = claimedBitMap[claimedWordIndex];
-        uint256 mask = (1 << claimedBitIndex);
-        return (claimedWord & mask) == mask;
+        (uint256 wordPos, uint256 bitPos) = _bitmapPositions(index);
```
Comment (krlosMata) [inline contracts/PolygonZkEVMBridge.sol:581]: maybe is worth it to take a look at https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/BitMaps.sol

Comment (krlosMata) [reply contracts/PolygonZkEVMBridge.sol:581]: as pointed by @invocamanman, `_setAndCheck` could be added in oppenzeppelin library

File: contracts/PolygonZkEVMBridge.sol [hunk 2]
```diff
@@ -587,35 +578,29 @@ contract PolygonZkEVMBridge is
...
+        uint256 mask = 1 << bitPos;
+        uint256 flipped = claimedBitMap[wordPos] ^= mask;
+        require (flipped & mask != 0,"PolygonZkEVMBridge::_setAndCheckClaimed: Already claimed");
     }
 
     /**
-     * @notice Function to extract the selector of a bytes calldata
-     * @param _data The calldata bytes
+     * @notice Function decode a index into a wordPos and bitPos
```
Comment (krlosMata) [inline contracts/PolygonZkEVMBridge.sol:598]:
```suggestion
     * @notice Function decode an index into a wordPos and bitPos
```

**Review by krlosMata — APPROVED** [review]
:smile_cat:

**Pull request merged**